BasicCalculator.cpp

```diff
@@ -0,0 +1,43 @@
+// Time COmplexity -> O(n)
+// Space Complexity -> O(n)
+// Problems Faced - It did not work if I took 'num' as 'int'. So I changed it to 'long'.
```

Comment:
Maybe in C++, It gives out of range for int, In Java if you use Stack of type Integer, it works fine with int .